app/models/service_template_ansible_playbook.rb

```diff
@@ -44,8 +44,9 @@ def self.create_catalog_item(options, auth_user)
           next unless dialog_name
 
           job_template = enhanced_config.fetch_path(action, :configuration_template)
-          enhanced_config[action][:dialog] =
-            Dialog::AnsiblePlaybookServiceDialog.create_dialog(dialog_name, job_template)
+          new_dialog = Dialog::AnsiblePlaybookServiceDialog.create_dialog(dialog_name, job_template)
+          enhanced_config[action][:dialog] = new_dialog
+          service_template.options[:config_info][action][:dialog_id] = new_dialog.id
         end
```

Comment:
@syncrou I have given it a lot of thoughts but still prefer the way as is. Line 48 and 49 use the result from Line 47. They are two very different hashes. If we pass them into a method as arguments and modify the hash content, it will be even harder to read or realize the hash content has been modified after the method call.